tests/std/tests/P0896R4_views_take_while/test.cpp

```diff
@@ -0,0 +1,493 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <forward_list>
+#include <ranges>
+#include <span>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+
+#include <iostream>
+
+// Test a silly precomposed range adaptor pipeline
+constexpr auto is_less_than_three = [](const auto& x) { return x < 3; };
+using Pred                        = remove_const_t<decltype(is_less_than_three)>;
+STATIC_ASSERT(is_nothrow_copy_constructible_v<Pred>&& is_nothrow_move_constructible_v<Pred>);
+
+constexpr auto pipeline = views::take_while(is_less_than_three) | views::take_while(is_less_than_three)
+                          | views::take_while(is_less_than_three) | views::take_while(is_less_than_three);
+
+template <class Rng, class V = views::all_t<Rng>>
+using pipeline_t = ranges::take_while_view<
+    ranges::take_while_view<ranges::take_while_view<ranges::take_while_view<V, Pred>, Pred>, Pred>, Pred>;
+
+template <class Rng>
+concept CanViewTake_while = requires(Rng&& r) {

```

Comment:
🤮🤮🤮 Could we maybe name this `CanViewTakeWhile`? 🤮🤮🤮

Reply:
:+1: